src/fp.c

```diff
@@ -570,6 +571,26 @@ void fp_clippy_trainer(void) {
     }
 }
 
+void fp_action_command_trainer(void) {
+    // Either goombario or mario attacking
+    if ((pm_battle_state_2 == 3 && pm_player.player_data.current_partner == PARTNER_GOOMBARIO) ||
+        pm_battle_state_2 == 4) {
+        if (pm_ActionCommandStatus.state == 10 && pm_status.pressed.a) {
+            fp.last_a_press = pm_status.frame_counter;
+        } else if (pm_ActionCommandStatus.state == 11) {
+            if (fp.last_a_press) {
+                fp_log("pressed A %d frames early", pm_status.frame_counter - fp.last_a_press);
```

Comment:
This has the issue of saying "pressed A 1 frames early". Personally I prefer using ternary operators for plurals like this, so you might do something like:
```c
int frames_early = pm_status.frame_counter - fp.last_a_press;
fp_log("pressed A %d frame%s early", frames_early, frames_early > 1 ? "s" : "");
```

Though if you know of a better way to do it, by all means.